Test scope dependencies missing in the generated BOM

With the below snippet the expectation was that junit dependency would be shown in the generated BOM, but noticed that it is missing. Notice that includeTestScope=true in the configuration below.

```
<dependencies>
	<dependency>
		<groupId>junit</groupId>
		<artifactId>junit</artifactId>
		<version>4.11</version>
		<scope>test</scope>
	</dependency>
	<dependency>
		<groupId>javax.servlet</groupId>
		<artifactId>javax.servlet-api</artifactId>
		<version>3.1.0</version>
		<scope>provided</scope>
	</dependency>
</dependencies>
<build>
	<plugins>
		<plugin>
			<groupId>org.cyclonedx</groupId>
			<artifactId>cyclonedx-maven-plugin</artifactId>
			<version>2.0.3</version>
			<executions>
				<execution>
					<phase>verify</phase>
					<goals>
						<goal>makeAggregateBom</goal>
					</goals>
				</execution>
			</executions>
			<configuration>
				<projectType>library</projectType>
				<schemaVersion>1.2</schemaVersion>
				<includeBomSerialNumber>true</includeBomSerialNumber>
				<includeCompileScope>true</includeCompileScope>
				<includeProvidedScope>true</includeProvidedScope>
				<includeRuntimeScope>true</includeRuntimeScope>
				<includeSystemScope>true</includeSystemScope>
				<includeTestScope>true</includeTestScope>
				<includeLicenseText>true</includeLicenseText>
				<outputFormat>xml</outputFormat>
			</configuration>
		</plugin>
	</plugins>
</build>
```

I tweaked plugin configuration in the POM so that phase was changed to `test`.   This resulted in production of a bom that still contains no test dependencies.  I also tested with `cyclonedx-cli`:
* diff  (comparing v2.4.1 bom with that generated using phase `verify`) reports that no component versions changed.
* validate reports that all seems good.  

However,  even though a valid bom was generated (albeit missing the test dependencies), the log now has an exception:

```
[INFO] --- cyclonedx-maven-plugin:2.4.1:makeBom (cyclonedx-aggregate) @ xxx ---
[INFO] CycloneDX: Parameters
[INFO] ------------------------------------------------------------------------
[INFO] schemaVersion          : 1.2
[INFO] includeBomSerialNumber : true
[INFO] includeCompileScope    : true
[INFO] includeProvidedScope   : true
[INFO] includeRuntimeScope    : true
[INFO] includeTestScope       : true
[INFO] includeSystemScope     : true
[INFO] includeLicenseText     : false
[INFO] outputFormat           : xml
[INFO] outputName             : bom
[INFO] ------------------------------------------------------------------------
[INFO] CycloneDX: Resolving Dependencies
[ERROR] Error encountered calculating hashes
java.io.IOException: Is a directory
    at sun.nio.ch.FileDispatcherImpl.read0 (Native Method)
    at sun.nio.ch.FileDispatcherImpl.read (FileDispatcherImpl.java:46)
    at sun.nio.ch.IOUtil.readIntoNativeBuffer (IOUtil.java:223)
    at sun.nio.ch.IOUtil.read (IOUtil.java:197)
    at sun.nio.ch.FileChannelImpl.read (FileChannelImpl.java:159)
    at sun.nio.ch.ChannelInputStream.read (ChannelInputStream.java:65)
    at sun.nio.ch.ChannelInputStream.read (ChannelInputStream.java:109)
    at sun.nio.ch.ChannelInputStream.read (ChannelInputStream.java:103)
    at org.apache.commons.codec.digest.DigestUtils.updateDigest (DigestUtils.java:1459)
    at org.apache.commons.codec.digest.DigestUtils.digest (DigestUtils.java:120)
    at org.apache.commons.codec.digest.DigestUtils.md5 (DigestUtils.java:482)
    at org.apache.commons.codec.digest.DigestUtils.md5Hex (DigestUtils.java:518)
    at org.cyclonedx.util.BomUtils.calculateHashes (BomUtils.java:53)
    at org.cyclonedx.maven.BaseCycloneDxMojo.convert (BaseCycloneDxMojo.java:443)
    at org.cyclonedx.maven.CycloneDxMojo.execute (CycloneDxMojo.java:64)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.jvnet.hudson.maven3.launcher.Maven35Launcher.main (Maven35Launcher.java:130)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at jenkins.maven3.agent.Maven35Main.launch (Maven35Main.java:178)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at hudson.maven.Maven3Builder.call (Maven3Builder.java:139)
    at hudson.maven.Maven3Builder.call (Maven3Builder.java:70)
    at hudson.remoting.UserRequest.perform (UserRequest.java:211)
    at hudson.remoting.UserRequest.perform (UserRequest.java:54)
    at hudson.remoting.Request$2.run (Request.java:376)
    at hudson.remoting.InterceptingExecutorService.lambda$wrap$0 (InterceptingExecutorService.java:78)
    at java.util.concurrent.FutureTask.run (FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:624)
    at java.lang.Thread.run (Thread.java:748)
[ERROR] An error occurred attempting to extract POM from artifact
java.io.FileNotFoundException: /home/jenkins/workspace/xxx-BOM/target/classes (Is a directory)
    at java.util.zip.ZipFile.open (Native Method)
    at java.util.zip.ZipFile.<init> (ZipFile.java:230)
    at java.util.zip.ZipFile.<init> (ZipFile.java:160)
    at java.util.jar.JarFile.<init> (JarFile.java:168)
    at java.util.jar.JarFile.<init> (JarFile.java:132)
    at org.cyclonedx.maven.BaseCycloneDxMojo.extractPom (BaseCycloneDxMojo.java:686)
    at org.cyclonedx.maven.BaseCycloneDxMojo.convert (BaseCycloneDxMojo.java:455)
    at org.cyclonedx.maven.CycloneDxMojo.execute (CycloneDxMojo.java:64)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.jvnet.hudson.maven3.launcher.Maven35Launcher.main (Maven35Launcher.java:130)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at jenkins.maven3.agent.Maven35Main.launch (Maven35Main.java:178)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at hudson.maven.Maven3Builder.call (Maven3Builder.java:139)
    at hudson.maven.Maven3Builder.call (Maven3Builder.java:70)
    at hudson.remoting.UserRequest.perform (UserRequest.java:211)
    at hudson.remoting.UserRequest.perform (UserRequest.java:54)
    at hudson.remoting.Request$2.run (Request.java:376)
    at hudson.remoting.InterceptingExecutorService.lambda$wrap$0 (InterceptingExecutorService.java:78)
    at java.util.concurrent.FutureTask.run (FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:624)
    at java.lang.Thread.run (Thread.java:748)
[INFO] CycloneDX: Creating BOM
[INFO] CycloneDX: Writing BOM (XML): /home/jenkins/workspace/xxx-BOM/target/bom.xml
[INFO] CycloneDX: Validating BOM (XML): /home/jenkins/workspace/xxx-BOM/target/bom.xml
[INFO] 
```

As an aside...  no logging of `projectType` ?  (I can log a separate issue for that if need be).

--- a/src/main/java/org/cyclonedx/maven/BaseCycloneDxMojo.java
+++ b/src/main/java/org/cyclonedx/maven/BaseCycloneDxMojo.java
@@ -744,7 +744,7 @@ public abstract class BaseCycloneDxMojo extends AbstractMojo implements Contextu
         return null;
     }
 
-    protected void execute(Set<Component> components, Set<Dependency> dependencies, MavenProject mavenProject) throws MojoExecutionException{
+    protected void execute(Set<Component> components, Set<Dependency> dependencies, MavenProject mavenProject) throws MojoExecutionException {
         try {
             getLog().info(MESSAGE_CREATING_BOM);
             final Bom bom = new Bom();

--- a/src/main/java/org/cyclonedx/maven/CycloneDxAggregateMojo.java
+++ b/src/main/java/org/cyclonedx/maven/CycloneDxAggregateMojo.java
@@ -20,6 +20,7 @@ package org.cyclonedx.maven;
 
 import org.apache.maven.artifact.Artifact;
 import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Execute;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
 import org.apache.maven.plugins.annotations.ResolutionScope;
@@ -39,9 +40,10 @@ import java.util.Set;
         defaultPhase = LifecyclePhase.PACKAGE,
         aggregator = true,
         requiresOnline = true,
-        requiresDependencyCollection = ResolutionScope.COMPILE_PLUS_RUNTIME,
-        requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME
+        requiresDependencyCollection = ResolutionScope.TEST,
+        requiresDependencyResolution = ResolutionScope.TEST
 )
+@Execute( phase = LifecyclePhase.TEST_COMPILE )
 public class CycloneDxAggregateMojo extends BaseCycloneDxMojo {
 
     protected boolean shouldExclude(MavenProject mavenProject) {

--- a/src/main/java/org/cyclonedx/maven/CycloneDxMojo.java
+++ b/src/main/java/org/cyclonedx/maven/CycloneDxMojo.java
@@ -20,6 +20,7 @@ package org.cyclonedx.maven;
 
 import org.apache.maven.artifact.Artifact;
 import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Execute;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
 import org.apache.maven.plugins.annotations.ResolutionScope;
@@ -33,9 +34,10 @@ import java.util.Set;
         name = "makeBom",
         defaultPhase = LifecyclePhase.PACKAGE,
         requiresOnline = true,
-        requiresDependencyCollection = ResolutionScope.COMPILE_PLUS_RUNTIME,
-        requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME
+        requiresDependencyCollection = ResolutionScope.TEST,
+        requiresDependencyResolution = ResolutionScope.TEST
 )
+@Execute( phase = LifecyclePhase.TEST_COMPILE )
 public class CycloneDxMojo extends BaseCycloneDxMojo {
 
     public void execute() throws MojoExecutionException {

--- a/src/main/java/org/cyclonedx/maven/CycloneDxPackageMojo.java
+++ b/src/main/java/org/cyclonedx/maven/CycloneDxPackageMojo.java
@@ -20,6 +20,7 @@ package org.cyclonedx.maven;
 
 import org.apache.maven.artifact.Artifact;
 import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Execute;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
 import org.apache.maven.plugins.annotations.ResolutionScope;
@@ -37,9 +38,10 @@ import java.util.Set;
         threadSafe = true,
         aggregator = true,
         requiresOnline = true,
-        requiresDependencyCollection = ResolutionScope.COMPILE_PLUS_RUNTIME,
-        requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME
+        requiresDependencyCollection = ResolutionScope.TEST,
+        requiresDependencyResolution = ResolutionScope.TEST
 )
+@Execute( phase = LifecyclePhase.TEST_COMPILE )
 public class CycloneDxPackageMojo extends BaseCycloneDxMojo {
 
     protected boolean shouldInclude(MavenProject mavenProject) {

--- a/src/test/java/org/cyclonedx/maven/Issue64Test.java
+++ b/src/test/java/org/cyclonedx/maven/Issue64Test.java
@@ -64,7 +64,7 @@ public class Issue64Test {
     private static void assertFileContains(File basedir, String expectedFile, String expectedContent) throws IOException {
         assertFilesPresent(basedir, expectedFile);
         String bomContents = fileRead(new File(basedir, expectedFile), true);
-        assertTrue(String.format("%s contains %s", expectedFile, expectedContent), bomContents.contains("junit"));
+        assertTrue(String.format("%s contains %s", expectedFile, expectedContent), bomContents.contains(expectedContent));
     }
 
     // source: https://github.com/takari/takari-plugin-testing-project/blob/master/takari-plugin-testing/src/main/java/io/takari/maven/testing/AbstractTestResources.java#L103

--- a/src/test/java/org/cyclonedx/maven/Issue64Test.java
+++ b/src/test/java/org/cyclonedx/maven/Issue64Test.java
@@ -1,0 +1,89 @@
+package org.cyclonedx.maven;
+
+import io.takari.maven.testing.TestResources;
+import io.takari.maven.testing.executor.MavenExecutionResult;
+import io.takari.maven.testing.executor.MavenRuntime;
+import io.takari.maven.testing.executor.MavenRuntime.MavenRuntimeBuilder;
+import io.takari.maven.testing.executor.MavenVersions;
+import io.takari.maven.testing.executor.junit.MavenJUnitTestRunner;
+import org.apache.commons.io.FileUtils;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.*;
+import java.util.Properties;
+
+import static io.takari.maven.testing.TestResources.assertFilesPresent;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(MavenJUnitTestRunner.class)
+@MavenVersions({"3.6.3"})
+public class Issue64Test {
+
+    @Rule
+    public final TestResources resources = new TestResources(
+            "target/test-classes",
+            "target/test-classes/transformed-projects"
+    );
+
+    public final MavenRuntime verifier;
+
+    public Issue64Test(MavenRuntimeBuilder runtimeBuilder)
+            throws Exception {
+        this.verifier = runtimeBuilder.build();
+    }
+
+    @Test
+    public void testPluginWithActiviti() throws IOException, Exception {
+        File projectDirTransformed = new File(
+                "target/test-classes/transformed-projects/issue-64"
+        );
+        if (projectDirTransformed.exists()) {
+            FileUtils.cleanDirectory(projectDirTransformed);
+            projectDirTransformed.delete();
+        }
+
+        File projDir = resources.getBasedir("issue-64");
+
+        Properties props = new Properties();
+
+        props.load(Issue117Test.class.getClassLoader().getResourceAsStream("test.properties"));
+        String projectVersion = String.class.cast(props.get("project.version"));
+        verifier
+                .forProject(projDir) //
+                .withCliOption("-Dtest.input.version=" + projectVersion) // debug
+                .withCliOption("-X") // debug
+                .withCliOption("-B")
+                .execute("clean", "verify")
+                .assertErrorFreeLog();
+
+        assertFileContains(projDir, "target/bom.xml", "junit");
+    }
+
+    private static void assertFileContains(File basedir, String expectedFile, String expectedContent) throws IOException {
+        assertFilesPresent(basedir, expectedFile);
+        String bomContents = fileRead(new File(basedir, expectedFile), true);
+        assertTrue(String.format("%s contains %s", expectedFile, expectedContent), bomContents.contains("junit"));
+    }
+
+    // source: https://github.com/takari/takari-plugin-testing-project/blob/master/takari-plugin-testing/src/main/java/io/takari/maven/testing/AbstractTestResources.java#L103
+    private static String fileRead(File file, boolean normalizeEOL) throws IOException {
+        StringBuilder sb = new StringBuilder();
+        try (BufferedReader r = new BufferedReader(new InputStreamReader(new FileInputStream(file)))) {
+            if (normalizeEOL) {
+                String str;
+                while ((str = r.readLine()) != null) {
+                    sb.append(str).append('\n');
+                }
+            } else {
+                int ch;
+                while ((ch = r.read()) != -1) {
+                    sb.append((char) ch);
+                }
+            }
+        }
+        return sb.toString();
+    }
+
+}

--- a/src/test/resources/issue-64/pom.xml
+++ b/src/test/resources/issue-64/pom.xml
@@ -45,7 +45,7 @@
             <plugin>
                 <groupId>org.cyclonedx</groupId>
                 <artifactId>cyclonedx-maven-plugin</artifactId>
-                <version>2.0.3</version>
+                <version>${test.input.version}</version>
                 <executions>
                     <execution>
                         <phase>verify</phase>

